### PR TITLE
sourcify Validation - APP-2101

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Added sourcify validation queries
- refactor sc address validation modal
- contract.ts script added
- sourcify net spec added to test it please use 0x1F98431c8aD98523631AE4a59f267346ea31F984 contract

Note: netspec comments will be added in next PR

Task: [AAP-2101](https://aragonassociation.atlassian.net/browse/APP-2101)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Review by RakeshUP — COMMENTED** [review]
Thanks a lot @sepehr2github. Looks and works great

**Review by RakeshUP — APPROVED** [review]

File: packages/web-app/src/utils/types.ts [hunk 1]
```diff
@@ -372,11 +372,32 @@ export type EtherscanContractResponse = {
   SourceCode: string;
 };
 
+export type SourcifyContractResponse = {
+  output: {
+    abi: SmartContractAction[];
+    devdoc: {
+      title: string;
+      methods: {
+        [key: string]: {
```
Comment (Fabricevladimir) [inline packages/web-app/src/utils/types.ts:382]: I would appreciate a couple comments on what the keys are

File: packages/web-app/src/containers/smartContractComposer/components/inputForm.tsx [hunk 2]
```diff
@@ -25,21 +25,20 @@ const InputForm: React.FC = () => {
       <p className="text-lg font-bold text-ui-800 capitalize">
         {selectedAction.name}
       </p>
-      <p className="mt-1 text-sm text-ui-600">
-        @notice as help text; This is the description of the method provided by
-        NatSpec Format or if those are our smart contracts, by further
-        implementation
-      </p>
+      <p className="mt-1 text-sm text-ui-600">{selectedAction.notice}</p>
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/smartContractComposer/components/inputForm.tsx]: Styled components please!

File: packages/web-app/src/containers/smartContractComposer/components/contractAddressValidation.tsx [hunk 3]
```diff
@@ -206,9 +370,9 @@ const ContractAddressValidation: React.FC<Props> = props => {
       />
       <Content>
         <DescriptionContainer>
-          <Title>{t('scc.addressValidation.title')}</Title>
+          <Title>{t('scc.validation.addressInputLabel')}</Title>
           <Description>
-            {t('scc.addressValidation.description')}{' '}
+            {t('scc.validation.addressInputHelp')}{' '}
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/smartContractComposer/components/contractAddressValidation.tsx]: Can you remove the empty space?

**Review by Fabricevladimir — APPROVED** [review]
Nice work! Left minor comments

**Comment by Fabricevladimir** [comment]
There's an issue when an unverified contract is cleared from the input. The calls seemingly are made with an empty value regardless. 
image

**Pull request merged**